contracts/secret-storage/SecretManager.sol

```diff
@@ -498,20 +514,28 @@ contract SecretManager is
         if(!userStorage[_secretId].selectedEnclaves[enclaveIndex].hasAcknowledgedStore)
             return;
 
-        _updateUsdcDepositPostDeadMark(_secretId, _enclaveAddress, enclaveIndex);
+        uint256 deadStorePayment = _getStoreUsdcPayment(_secretId, _enclaveAddress, enclaveIndex);
+        userStorage[_secretId].usdcDeposit -= deadStorePayment;
```

Comment:
this should be subtracted for secrets only which are terminated or end time has passed?

Reply:
Also even then you are just subtracting it from user's account. You should add this removed value to some other state variable representing the USDC that contract owner can withdraw.

Reply:
It can be called at anytime to mark the store dead. 
But for alive check, it will be executed post termination only. Same for remove secret, to mark the stores dead.